### PR TITLE
Docs: Plugin doc review changes from chunk 1-B with corrected set of files

### PR DESCRIPTION
**Pull request by josmperez**
**What is this feature?**

This PR replaces/corrects the second of two PRs holding a review of plugins docs. The file it replaces unfortunately included changes to files that were not part of the PR. 

**Why do we need this feature?**

This fulfills the doc improvement initiative described in the plugins doc plan.

**Who is this feature for?**

Plugin developers

### CODE REVIEW

**Review by andresmgot — APPROVED** [review]
Content looks good to me :+1: 

You also have some lint errors here though: https://drone.grafana.net/grafana/grafana/112914/1/5

File: docs/sources/developers/plugins/package-a-plugin.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+description: How to package a plugin
+title: Package a plugin
+---
+
+You've just built your first plugin and you're getting ready to [publish it](publish-a-plugin.md), and now you need to package it. This step follows the building and [signing](sign-a-plugin.md) steps for plugin deployment.
```
Comment (marefr) [inline docs/sources/developers/plugins/package-a-plugin.md:6]: Shouldn't `relref` be used here or isn't it needed when in the same directory? There are some other similar changes where this is applicable. Ref https://grafana.com/docs/writers-toolkit/writing-guide/references/

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 2]
```diff
@@ -7,140 +7,39 @@ title: Publish a plugin
...
-### Frequently Asked Questions
-
-**Do I need to submit a private plugin?**
-
-- No. Please only submit plugins that you wish to make publicly available for the Grafana community.
-
-**How long does it take to review my submission?**
-
-- We're not able to give an estimate at this time, though we're constantly working on improving the time it takes to review a plugin.
+# Publish or update a plugin
```
Comment (marefr) [inline docs/sources/developers/plugins/publish-a-plugin.md:10]: Would the alias used above `package-a-plugin` conflict with the new package-a-plugin.md file you've created?

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 3]
```diff
@@ -151,11 +50,11 @@ To speed up the time it takes to review your plugin:
      - Select **Multiple** if you'd like to submit separate plugin archives for each architecture.
        This can lead to faster downloads since users can select the specific architecture on which they want to install the plugin.
    - **URL:** A URL that points to a ZIP archive of your packaged plugin.
-   - **Source Code URL:** A URL that points to a public git repository or ZIP archive of your complete plugin source code.
+   - **Source Code URL:** A URL that points to a public Git repository or ZIP archive of your complete plugin source code.
```
Comment (marefr) [inline docs/sources/developers/plugins/publish-a-plugin.md:53]: I think we should provide some examples here. Seen many times when reviewing plugins that the provided source code URL not necessarily is connected to the git tag the plugin was built from. @academo you maybe can provide some more context/input here?  

This would also be applicable to update a plugin below.

**Review by marefr — COMMENTED** [review]
Looks very good. Added some comments/questions. Also asked for additional eyes on the data frame changes

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 3 above)
Comment (academo) [reply docs/sources/developers/plugins/publish-a-plugin.md:53]: I remember when we added this part to the documentation we thought it'd be too much to put there but I leave you to decide.

Valid URls examples:


Using a tag or branch: https://github.com/grafana/clock-panel/tree/v2.1.3
A tag or branch and the code is in a subdirectory (important for mono repos): https://github.com/grafana/clock-panel/tree/v2.1.3/plugin/ (here `plugin` contains the plugin code)
Using the latest main or master branch commit: https://github.com/grafana/clock-panel/ (not recommended, better pass a tag or branch)

Comment (marefr) [reply docs/sources/developers/plugins/publish-a-plugin.md:53]: Thanks. So maybe to limit the amount of details here add a question under `Frequently asked questions` like "What Source Code URL formats are supported" or something?

File: docs/sources/developers/plugins/data-frames.md [hunk 4]
```diff
@@ -6,15 +6,13 @@ title: Data frames
 
 Grafana supports a variety of different data sources, each with its own data model. To make this possible, Grafana consolidates the query results from each of these data sources into one unified data structure called a _data frame_.
 
-The data frame structure is a concept that's borrowed from data analysis tools like the [R programming language](https://www.r-project.org), and [Pandas](https://pandas.pydata.org/).
-
-> Data frames are available in Grafana 7.0+, and replaced the Time series and Table structures with a more generic data structure that can support a wider range of data types.
-
-This document gives an overview of the data frame structure, and of how data is handled within Grafana.
+> **Note:** Data frames are available in Grafana 7.0+, and replaced the Time series and Table structures with a more generic data structure that supports a wider range of data types.
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/data-frames.md:9]: At what stage can we just remove notes like this... We have versioned docs, and this is now a 3 year old notice?

File: docs/sources/developers/plugins/data-frames.md [hunk 5]
```diff
@@ -26,45 +24,47 @@ interface DataFrame {
 }
 ```
 
-In essence, a data frame is a collection of _fields_, where each field corresponds to a column. Each field, in turn, consists of a collection of values, along with meta information, such as the data type of those values.
+### Data frame fields
+
+In essence, a data frame is a collection of _fields_, where each field corresponds to a column. Each field, in turn, consists of a collection of values, along with metadata, such as the data type of those values.
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/data-frames.md:29]: Nit: "In essence" feels unnecessary

File: docs/sources/developers/plugins/package-a-plugin.md (hunk 1 above)
Comment (josmperez) [reply docs/sources/developers/plugins/package-a-plugin.md:6]: It may be, I will double-check and update/fix rel-refs.

File: docs/sources/developers/plugins/data-frames.md [hunk 6]
```diff
@@ -6,15 +6,13 @@ title: Data frames
...
 
 ## The data frame
 
-A data frame is a columnar-oriented table structure, which means it stores data by column and not by row. To understand what this means, let’s look at the TypeScript definition used by Grafana:
+Data frames have a columnar-oriented table structure, which means it stores data by column and not by row. 
+
+You can see what this means by looking at the TypeScript definition used by Grafana:
 
 ```ts
 interface DataFrame {
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/data-frames.md:18]: Is this out of date? https://github.com/grafana/grafana/blob/c308118fc024eae0a7bf48a0b681e234e37262c3/packages/grafana-data/src/types/dataFrame.ts#L236

File: docs/sources/developers/plugins/data-frames.md [hunk 7]
```diff
@@ -26,45 +24,47 @@ interface DataFrame {
 }
 ```
 
-In essence, a data frame is a collection of _fields_, where each field corresponds to a column. Each field, in turn, consists of a collection of values, along with meta information, such as the data type of those values.
+### Data frame fields
+
+In essence, a data frame is a collection of _fields_, where each field corresponds to a column. Each field, in turn, consists of a collection of values, along with metadata, such as the data type of those values.
 
 ```ts
 interface Field {
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/data-frames.md:32]: https://github.com/grafana/grafana/blob/c308118fc024eae0a7bf48a0b681e234e37262c3/packages/grafana-data/src/types/dataFrame.ts#L126 similarly may be out of date?

File: docs/sources/developers/plugins/data-frames.md [hunk 8]
```diff
@@ -26,45 +24,47 @@ interface DataFrame {
...
 
-### Field configuration
+### Field configurations
 
 Each field in a data frame contains optional information about the values in the field, such as units, scaling, and so on.
 
 By adding field configurations to a data frame, Grafana can configure visualizations automatically. For example, you could configure Grafana to automatically set the unit provided by the data source.
 
-## Transformations
+## Data transformations
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/data-frames.md:65]: Maybe we can link to https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/ and ensure we're still up to date here

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 2 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:10]: Good question. I think I need to remove the alias, but I'm double-checking now.

File: docs/sources/developers/plugins/data-frames.md [hunk 9]
```diff
@@ -146,7 +146,7 @@ Dimensions: 4 fields by 4 rows
 +---------------------+-----------------+-----------------+----------------+
 ```
 
-can be converted into a data frame in wide format:
+Convert the above table into a data frame in wide format like this:
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/data-frames.md:149]: How does the user convert the data format? Or is this this illustrative of the end result in which case the text should read closer to the original

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 3 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:51]: I have added these examples as one of the FAQs and added a link from the procedural documentation.

File: docs/sources/developers/plugins/data-frames.md (hunk 4 above)
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:9]: I think it's safe to remove, but I also think you asked a great question about versioning notes relevancy and I'm looking into this.

File: docs/sources/developers/plugins/data-frames.md (hunk 5 above)
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:29]: I'm a fan of concision. Deleted.

File: docs/sources/developers/plugins/data-frames.md (hunk 6 above)
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md]: Good catch, I've commited a fix.

File: docs/sources/developers/plugins/data-frames.md (hunk 7 above)
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md]: Good catch, I've commited a fix.

File: docs/sources/developers/plugins/data-frames.md (hunk 8 above)
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:97]: Good catch, I've commited a fix.

File: docs/sources/developers/plugins/data-frames.md (hunk 9 above)
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md]: Good catch, I've commited a fix.

File: docs/sources/developers/plugins/package-a-plugin.md (hunk 1 above)
Comment (jdbaldry) [reply docs/sources/developers/plugins/package-a-plugin.md:6]: Note: latest advice is in https://github.com/grafana/writers-toolkit/pull/225.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 2 above)
Comment (jdbaldry) [reply docs/sources/developers/plugins/publish-a-plugin.md:10]: Aliases only result in a redirect page being created if there isn't already a page there, documented in https://grafana.com/docs/writers-toolkit/writing-guide/front-matter/#hugo-aliases.

I would still remove the alias for cleanliness.

File: docs/sources/developers/plugins/package-a-plugin.md (hunk 1 above)
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/package-a-plugin.md:6]: You might not want to publish, you might just be looking to package and use locally.

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 10]
```diff
@@ -7,140 +7,39 @@ title: Publish a plugin
...
-
-- No. Please only submit plugins that you wish to make publicly available for the Grafana community.
-
-**How long does it take to review my submission?**
-
-- We're not able to give an estimate at this time, though we're constantly working on improving the time it takes to review a plugin.
+# Publish or update a plugin
 
-**Can I decide a date when my plugin will be published?**
+You've just built your first plugin and [packaged it](package-a-plugin.md), and now you want to share it with the world.
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/publish-a-plugin.md:12]: this link 404s for me on a local build as it tries to get to http://localhost:3002/docs/grafana/latest/developers/plugins/publish-a-plugin/package-a-plugin.md  rather than the correct - http://localhost:3002/docs/grafana/latest/developers/plugins/package-a-plugin/

Comment (sympatheticmoose) [reply docs/sources/developers/plugins/publish-a-plugin.md:12]: As in the previous PR, I would like to remove notions of this being your first plugin.

File: docs/sources/developers/plugins/package-a-plugin.md (hunk 1 above)
Comment (josmperez) [reply docs/sources/developers/plugins/package-a-plugin.md:6]: @jdbaldry Good to know. I probably missed it, but I've looked in the writer's toolkit and don't see any explicit examples of links to files in the same directory. Chris confirmed that relrefs are needed there, so I've added them in.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 2 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:10]: Removed.

File: docs/sources/developers/plugins/data-frames.md (hunk 4 above)
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:9]: Removed.

File: docs/sources/developers/plugins/error-handling.md [hunk 11]
```diff
@@ -4,7 +4,7 @@ title: Error handling
 
 # Error handling
 
-This guide explains how to handle errors in plugins.
+This guide explains how to handle errors in plugins and provides suggestions for common scenarios.
 
 ## Provide usable defaults
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/error-handling.md:9]: When phase 2 is complete, this page will live in Create a plugin. The error (my bad) was not retitling this page to 'Work with error messages'. Alternatively, we could divide the page in two and put one in conceptual and one in how-to, but the page really isn't that long.

File: docs/sources/developers/plugins/package-a-plugin.md (hunk 1 above)
Comment (josmperez) [reply docs/sources/developers/plugins/package-a-plugin.md:6]: ok

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 10 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:12]: @jdbaldry While fixing a relref in the doc, I identified three places where a "Sign in" link to Grafana Cloud was given with a relative path and no relref. I fixed it, but want to check to see if there's something special about Cloud's sign in that would make it an exception?

Comment (jdbaldry) [reply docs/sources/developers/plugins/publish-a-plugin.md]: The destinations in https://github.com/grafana/grafana/pull/67164/commits/663481ba34bf0afa7db8b371e40d3a08d4ef6846#diff-0d9267ff4b6486865c95041d2198385726213f3fd9b8736900104c49ca8359a8L42 are actually absolute URL paths because of the preceding slash (`/`). A relative path would start with one or two periods (`./` or `../`).

I realized the updated documentation doesn't clearly explain when to use relrefs, so I added a new note: https://github.com/grafana/grafana/pull/67164/commits/663481ba34bf0afa7db8b371e40d3a08d4ef6846#diff-0d9267ff4b6486865c95041d2198385726213f3fd9b8736900104c49ca8359a8L42.

In this case, because the link is external to the project documentation, it should not be a relref.

File: docs/sources/developers/plugins/package-a-plugin.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+-rwxr-xr-x   15209472 bX defN 22-Mar-24 23:23 grafana-clickhouse-datasource/gpx_clickhouse_windows_amd64.exe
+drwxr-xr-x          0 bx stor 22-Mar-24 23:23 grafana-clickhouse-datasource/img/
+-rw-r--r--        304 bX defN 22-Mar-24 23:23 grafana-clickhouse-datasource/img/logo.png
+-rw-r--r--       1587 bX defN 22-Mar-24 23:23 grafana-clickhouse-datasource/img/logo.svg
+-rw-r--r--     138400 bX defN 22-Mar-24 23:23 grafana-clickhouse-datasource/module.js
+-rw-r--r--        808 bX defN 22-Mar-24 23:23 grafana-clickhouse-datasource/module.js.LICENSE.txt
+-rw-r--r--     487395 bX defN 22-Mar-24 23:23 grafana-clickhouse-datasource/module.js.map
+-rw-r--r--       1616 bX defN 22-Mar-24 23:23 grafana-clickhouse-datasource/plugin.json
+22 files, 92516655 bytes uncompressed, 34319591 bytes compressed:  62.9%
+```
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/package-a-plugin.md:65]: This ends fairly abruptly, perhaps we could have a next steps section which links to publish as well as install http://localhost:3002/docs/grafana/latest/administration/plugin-management/#install-a-packaged-plugin

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 13]
```diff
@@ -7,140 +7,39 @@ title: Publish a plugin
...
-
-- No. We will not accept any new plugin submissions written in angular.
-  Please take a look at our [angular support deprecation documentation]({{< relref "../angular_deprecation" >}}).
-
-### Publishing your plugin for the first time
+To speed up the time it takes to review your plugin:
 
-**Do plugin signatures expire?**
+- Check that your plugin is ready for review using the [plugin validator](https://github.com/grafana/plugin-validator).
+- Read our [6 tips for improving your Grafana plugin before you publish](/blog/2021/01/21/6-tips-for-improving-your-grafana-plugin-before-you-publish/).
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/publish-a-plugin.md:34]: This page is quite out of date and links to deprecated tech. Suggest we remove for now and have a follow up issue to update and bring this content into the main docs

File: docs/sources/developers/plugins/working-with-data-frames.md [hunk 14]
```diff
@@ -1,20 +1,20 @@
 ---
-title: Working with data frames
+title: Work with data frames
 ---
 
-# Working with data frames
+# Work with data frames
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/working-with-data-frames.md:5]: Working feels more correct here, interested to know why the change?

Why don't we just merge this content into data-frames.md?

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 15]
```diff
@@ -1,148 +1,44 @@
...
 title: Publish a plugin
 type: docs
 ---
 
-# Publish a plugin
+# Publish or update a plugin
 
-You've just built your first plugin, and now you want to share it with the world.
-In this guide, you'll learn how to package and share your plugin with others.
+You've just built your first plugin and [packaged it]({{< relref "package-a-plugin.md" >}}), and now you want to share it with the world.
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/publish-a-plugin.md:11]: do we want to link "just built" to `_index.md#tutorials`?

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 16]
```diff
@@ -1,148 +1,44 @@
...
+The best way to share your plugin with the world is to publish it in the [Grafana plugin catalog]({{< "/plugins" >}}).
+By having your plugin published on Grafana.com, more users will be able to discover your plugin.
 
-1. (Optional) Verify your plugin is packaged correctly using [zipinfo](https://linux.die.net/man/1/zipinfo).
-   It should look like this:
+## Before you begin
 
-```
-zipinfo grafana-clickhouse-datasource-1.1.2.zip
+Before you begin to publish a plugin, it needs to be built and packaged.
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/publish-a-plugin.md:19]: "it needs to be built and packaged" and the first sentence in the section above feel a bit redundant?

Comment (tolzhabayev) [reply docs/sources/developers/plugins/publish-a-plugin.md:19]: same with package as we say "you've just built your first plugin and packaged it" already at the top?

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 17]
```diff
@@ -1,148 +1,44 @@
...
+## Publish your plugin
 
-### Publishing your plugin for the first time
+Follow these steps to publish your plugin for the first time.
 
-**Do plugin signatures expire?**
-
-- Plugin signatures do not currently expire.
-
-{{< figure src="/static/img/docs/plugins/plugins-submission-create2.png" class="docs-image--no-shadow" max-width="650px" >}}
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/publish-a-plugin.md:143]: looks like we are removing the screenshot of the process, is it on purpose?

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 18]
```diff
@@ -180,6 +76,53 @@ To submit an **update** for an already published plugin:
...
+
+### Do I need to submit a private plugin?
+
+- No. Please only submit plugins that you wish to make publicly available for the Grafana community.
+
+### How long does it take to review my submission?
+
+- We're not able to give an estimate at this time, though we're constantly working on improving the time it takes to review a plugin.
+
+### an I decide a date when my plugin will be published?
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/publish-a-plugin.md:93]:
```suggestion
### Can I decide a date when my plugin will be published?
```

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 19]
```diff
@@ -180,6 +76,53 @@ To submit an **update** for an already published plugin:
...
+
+### Can I unlist my plugin from the Grafana plugin catalog in case of a bug?
+
+- In the event of a bug, unlisting the plugin from our catalog may be possible in exceptional cases, such as security concerns. However, we don't have control over the specific instances where the plugin is installed.
+
+### Can I distribute my plugin somewhere else other than the Grafana plugin catalog?
+
+- The official method for distributing Grafana plugins is through our catalog. Alternative methods, such as installing private or development plugins on local Grafana instances, are available as per the guidelines provided in [this guide]({{< relref "../../administration/plugin-management#install-plugin-on-local-grafana" >}}).
+
+### an I still use Angular for my plugin?
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/publish-a-plugin.md:114]:
```suggestion
### Can I still use Angular for my plugin?
```

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 20]
```diff
@@ -180,6 +76,53 @@ To submit an **update** for an already published plugin:
...
+
+- No. We will not accept any new plugin submissions written in angular. For more information, refer to our [Angular support deprecation documentation]({{< relref "../angular_deprecation" >}}).
+
+### Do plugin signatures expire?
+
+- Plugin signatures do not currently expire.
+
+{{< figure src="/static/img/docs/plugins/plugins-submission-create2.png" class="docs-image--no-shadow" max-width="650px" >}}
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/publish-a-plugin.md:122]: this image moved down here seems like a wrong place for it as it it's not really related to plugin signature expiration

File: docs/sources/developers/plugins/publish-a-plugin.md [hunk 21]
```diff
@@ -180,6 +76,53 @@ To submit an **update** for an already published plugin:
...
        This can lead to faster downloads since users can select the specific architecture they want to install the plugin on.
    - **URL:** A URL that points to a ZIP archive of your packaged plugin.
-   - **Source Code URL:** A URL that points to a public git repository or ZIP archive of your complete plugin source code.
+   - **Source Code URL:** A URL that points to a public Git repository or ZIP archive of your complete plugin source code. See [examples](#what-source-code-url-formats-are-supported).
    - **MD5:** The MD5 hash of the plugin specified by the **URL**.
 1. Click **Submit**.
+
+## Frequently asked questions
+
+### Do I need to submit a private plugin?
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/publish-a-plugin.md:85]: Feel like the titles and ordering of the FAQ could use some extra love, but we can also do it in any other phase

**Review by tolzhabayev — DISMISSED** [review]
smaller changes + see all the other feedback

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 10 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md]: Ok, will check in fix.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 13 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:34]: Ok. Created https://github.com/grafana/grafana/issues/67545.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 15 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:11]: Added link to "/create-a-grafana-plugin/"

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 16 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md]: Fix checked in.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 18 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md]: Fix checked in.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 19 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md]: Fix checked in.

File: docs/sources/developers/plugins/package-a-plugin.md (hunk 1 above)
Comment (jdbaldry) [reply docs/sources/developers/plugins/package-a-plugin.md]: LGTM, thanks!

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 20 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md:121]: Thanks, fixed it.

File: docs/sources/developers/plugins/migration-guide/angular-react/_index.md [hunk 22]
```diff
@@ -13,31 +13,17 @@ weight: 1000
...
 
-If you’re looking to migrate a plugin to the new plugin platform, then we recommend that you release it under a new major version. Consider keeping a release branch for the previous version to be able to roll out patch releases for versions prior to Grafana 7.
+If you want to migrate a plugin to Grafana's React-based plugin platform, then we recommend that you release it under a new major version. Consider keeping a release branch for the previous version to be able to roll out patch releases for versions prior to Grafana 7.
 
-While there's no 1-to-1 migration path from an Angular plugin to the new React platform, from early adopters, we’ve learned that one of the easiest ways to migrate is to:
+While there's no standard migration path from an Angular plugin to the new React platform, we’ve learned that one of the easiest ways to migrate is to:
 
 1. Create a new branch called `migrate-to-react`.
-1. Start from scratch with one of the templates provided by Grafana Toolkit.
+1. Start from scratch with one of the templates provided by the Grafana Toolkit. SHOULDN'T THIS BE THE CREATE PLUGIN TOOL?
```
Comment (tolzhabayev) [inline docs/sources/developers/plugins/migration-guide/angular-react/_index.md:21]: I think there is something that went wrong on this line. And yes it should be create-plugin tool

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 20 above)
Comment (tolzhabayev) [reply docs/sources/developers/plugins/publish-a-plugin.md:121]: I still see it in the same place as before. Maybe some changes were not pushed yet?

File: docs/sources/developers/plugins/working-with-data-frames.md [hunk 23]
```diff
@@ -1,20 +1,20 @@
...
 
-The data frame is a columnar data structure which allows efficient querying of large amounts of data. Since data frames are a central concept when developing plugins for Grafana, in this guide we'll look at some ways you can use them.
+The [data frame](data-frames.md) is a columnar data structure that allows for efficient querying of large amounts of data. Since data frames are a central concept when developing plugins for Grafana, in this guide we'll look at some ways you can use them.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/working-with-data-frames.md:7]:
```suggestion
The [data frame]({{< relref "data-frames" >}}) is a columnar data structure that allows for efficient querying of large amounts of data. Since data frames are a central concept when developing plugins for Grafana, in this guide, we'll look at some ways you can use them.
```

File: docs/sources/developers/plugins/working-with-data-frames.md [hunk 24]
```diff
@@ -57,7 +57,7 @@ function SimplePanel({ data: Props }) {
 }
 ```
 
-Before you start reading the data, think about what data you expect. For example, to visualize a time series we'd need at least one time field, and one number field.
+Before you start reading the data, think about what data you expect. For example, to visualize a time series we need at least one time field and one number field.
```
Comment (imatwawana) [inline docs/sources/developers/plugins/working-with-data-frames.md:60]:
```suggestion
Before you start reading the data, think about what data you expect. For example, to visualize a time series you need at least one time field and one number field.
```

**Review by imatwawana — COMMENTED** [review]
This mostly looks good and most of my suggestions are minor wording fixes. I have left a couple questions in spots where I was confused. I'd like to take another pass when you've had time to review just to make sure those parts are easily understood.

File: docs/sources/developers/plugins/publish-a-plugin.md (hunk 20 above)
Comment (josmperez) [reply docs/sources/developers/plugins/publish-a-plugin.md]: Pushed the fixed file.

File: docs/sources/developers/plugins/data-frames.md [hunk 25]
```diff
@@ -6,68 +6,100 @@ title: Data frames
...
+  /**
+   *  Field value type (string, number, etc)
+   */
   type: FieldType;
-  // Array of values all of the same type
-  values: Vector<T>;
-
-  // Optional display data for the field (e.g. unit, name over-ride, etc)
+  /**
+   *  Meta info about how field and how to display it
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:40]: not sure what you're pointing to @imatwawana

[the remark this answers is not in the document]
Comment (imatwawana) [reply docs/sources/developers/plugins/data-frames.md:40]: The first half of the sentence doesn't make sense: "Meta info about how field". Either there are words missing or maybe the word "how" shouldn't be there. But I wasn't clear what the intent was to make a suggestion.

[the remark this answers is not in the document]
Comment (josmperez) [reply docs/sources/developers/plugins/data-frames.md:40]: duh, got it

**Review by imatwawana — APPROVED** [review]
